adblock-lean

```diff
@@ -301,77 +310,80 @@ generate_preprocessed_blocklist_file_parts()
-
-					preprocessed_blocklist_line_count=$(( preprocessed_blocklist_line_count + blocklist_file_part_line_count ))
-					blocklist_id=$((blocklist_id+1))
-					continue 2
-				else
-					log_msg "Downloaded blocklist file part line count: ${blocklist_file_part_line_count} less than configured minimum: ${min_blocklist_file_part_line_count}."
-					break
-				fi
+			if [[ "${rogue_element_action}" != "IGNORE" ]]
+			then
+				log_msg "Checking for any rogue elements."
+				tee >(grep -Evn -m1 '^(local=/[[:alnum:]*][[:alnum:]*_.-]+/|bogus-nxdomain=[0-9.]+){0,1}$' > /tmp/rogue_element)
 			else
+				cat
+			fi | gzip > /tmp/blocklist.${blocklist_id}.gz
+
+			blocklist_file_part_size_KB=$(( ($(cat /tmp/blocklist_part_size_B 2>/dev/null) + 0) / 1024 ))
+			blocklist_file_part_line_count="$(cat /tmp/blocklist_part_line_count 2>/dev/null)"
+			: "${blocklist_file_part_line_count:=0}"
+
+			rm -f /tmp/blocklist_part_size_B /tmp/blocklist_part_line_count
+
+			if ! grep -q "Download completed" /tmp/uclient-fetch_err
+			then
 				log_msg "Download of new blocklist file part from: ${blocklist_url} failed."
-				if [[ -f "/tmp/blocklist.${blocklist_id}" ]]
+				if [[ "${blocklist_file_part_size_KB}" -ge "${max_blocklist_file_part_size_KB}" ]]
 				then
-					blocklist_file_part_size_KB=$(get_file_size_KB /tmp/blocklist.${blocklist_id})
-					if [[ "${blocklist_file_part_size_KB}" -eq "${max_blocklist_file_part_size_KB}" ]]
-					then
 						log_msg "Downloaded blocklist file part size exceeded the maximum value set in config (${max_blocklist_file_part_size_KB} KB)."
 						log_msg "Consider either increasing this value in the config or removing the correasponding blocklist url."
 						break
-					fi
 				fi
+				log_msg "Sleeping for 5 seconds after failed download attempt."
+				sleep 5
+				continue
+			fi
+
+			if read -r rogue_element < /tmp/rogue_element
+			then
+				log_msg "Rogue element: '${rogue_element}' identified originating in blocklist file part from: ${blocklist_url}."
+
+				if [[ "${rogue_element_action}" == "STOP" ]]
+				then
+					return 1
+				else
+					log_msg "Skipping file part and continuing."
+					rm -f /tmp/rogue_element
+					continue 2
+				fi
+			fi
+
+			if [[ "${blocklist_file_part_line_count}" -ge "${min_blocklist_file_part_line_count}" ]]
+			then
+				log_msg "Download of new blocklist file part from: ${blocklist_url} succeeded (downloaded file size: ${blocklist_file_part_size_KB} KB; sanitized line count: ${blocklist_file_part_line_count})."
+
+				rm -f /tmp/rogue_element
+
+				preprocessed_blocklist_line_count=$(( preprocessed_blocklist_line_count + blocklist_file_part_line_count ))
+				blocklist_id=$((blocklist_id+1))
+				continue 2
+			else
+				log_msg "Downloaded blocklist file part line count: ${blocklist_file_part_line_count} less than configured minimum: ${min_blocklist_file_part_line_count}."
+				break
 			fi
-			log_msg "Sleeping for 5 seconds after failed download attempt."
```

Comment:
Actually maybe we should keep these in just in case, as you suggested. But I still prefer to keep inline even with the duplication.